### PR TITLE
NLog - MessageTemplateParameter - IsReservedFormat

### PR DESCRIPTION
**Pull request by snakefoot**
Moved MessageTemplate UnitTests into the MessageTemplates-namespace.

### CODE REVIEW

**Comment by 304NotModified** [comment]
in what case would you like to use the IsReservedFormat?

File: src/NLog/MessageTemplates/MessageTemplateParameter.cs [hunk 1]
```diff
@@ -62,6 +62,28 @@ public struct MessageTemplateParameter
         public readonly string Format;
 
         /// <summary>
+        /// Checks if the <see cref="Format"/> contains reserved letters ('@', '$', 'l')
+        /// </summary>
+        public bool IsReservedFormat
```
Comment (304NotModified) [inline src/NLog/MessageTemplates/MessageTemplateParameter.cs]: Another option is to create a static method, so we don't have to construct a new MessageTemplateParameter to check the format

**Comment by snakefoot** [comment]
> in what case would you like to use the IsReservedFormat?

Was thinking that I wanted to help the Seq-engine. It wants to pass the parameters through its own custom formatter. So it wants to know if it is "$" or "@" or "l", but I also want to tell users they should not use the format-string together with string.Format.

File: src/NLog/MessageTemplates/MessageTemplateParameter.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/MessageTemplates/MessageTemplateParameter.cs]: It is mostly an informational courtesy for outsiders. No one but Seq, and a few others will be messing with the MessageTemplateParameter

Comment (snakefoot) [reply src/NLog/MessageTemplates/MessageTemplateParameter.cs]: And because I have been evil and merged CaptureType into Format

Comment (snakefoot) [reply src/NLog/MessageTemplates/MessageTemplateParameter.cs]: Since door was already open with the custom format "l" (lowercase L)

Comment (304NotModified) [reply src/NLog/MessageTemplates/MessageTemplateParameter.cs]: OK, but why not a static method? I think that's more clean.

Comment (304NotModified) [reply src/NLog/MessageTemplates/MessageTemplateParameter.cs]: Another option is a C# extension, benefit, it "pollutes" the class MessageTemplateParameter less.

Comment (snakefoot) [reply src/NLog/MessageTemplates/MessageTemplateParameter.cs]: Both the C# extension and the static-method that takes the struct as parameter will require stack-allocation of the MessageTemplateParameter. Not sure I understand the problem with a struct-property.

Comment (snakefoot) [reply src/NLog/MessageTemplates/MessageTemplateParameter.cs]: But we can make it a static-method that takes the format-string as input. Would that be better?

Comment (304NotModified) [reply src/NLog/MessageTemplates/MessageTemplateParameter.cs]: Well it's a bit odd here, but the struct reason is a good reason to have it like this.

**Comment by snakefoot** [comment]
@304NotModified If something should be changed, then please tell. Would prefer to have this included before nblumhardt does the testdrive with https://github.com/datalust/nlog-targets-seq

**Comment by 304NotModified** [comment]
merged!

**Pull request merged**